### PR TITLE
Fix Arithmetic Error

### LINKED ISSUE #3538: Disappering Gold Pans
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

Electric gold pans are disappearing if you interact with them too much.

### 📑 Reproduction Steps

1. Make an electric gold pan 3 process 
2. spam right click and ESC on it.

### 💡 Expected Behavior

Not to disappear.

### 📷 Screenshots / Videos

https://user-images.githubusercontent.com/85101965/164281649-2eb8e575-87a5-4139-b0a7-4e4c87607373.mp4

[Error Report 2](https://privatebin.net/?3fd0180b9d95cd98#7SGkpFZx5BSRDucviKhLHsZFeyiwb4bubgD2h78DHjFx)
[Error Report 3 - Different stacktrace](https://privatebin.net/?edcc64e2b376a177#HxkXrYJfP9mpzS1ZpNcKmbcprqyDhCDh9Lk4DGB7KuRB)

### 📜 Server Log

https://privatebin.net/?8ed346065e73474b#QniTSpzbUqcWtSNkEUuKgdZQ49yJwhZrVvEjTR5xHn7

### 📂 `/error-reports/` folder

https://privatebin.net/?3deb9cba446fd7fd#12i9wb6HDbwKwEfEfau7moK6STyvigPbpKs2e4SbKGT6

### 💻 Server Software

Purpur

### 🎮 Minecraft Version

1.18.x

### ⭐ Slimefun version

Screenshot_20220420_194851

### 🧭 Other plugins

_No response_

### PR DESCRIPTION
**Pull request by itslukemango**
## Description

We found this issue by accident, but it seems that if Slimefun's ticks are faster than expected, Tier 3 items such as Electric Ore Grinders will produce an ArithmeticException.

https://pastebin.com/ZGWDJyWn - the error from our fork, this small patch fixed it.

## Proposed changes

Don't update the progress bar if the operation is already finished

## Related Issues (if applicable)

Resolves #3538

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by Sfiguz7 — DISMISSED** [review]
It's likely Biscuit will want the update progress to be in an else block but I personally don't mind it. I will just approve and let him or Walshy decide if they want to suggest the change before merging

**Comment by itslukemango** [comment]
yeah I mean I don't mind changing it if that's what's preferred :)

**Review by svr333 — APPROVED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/machines/MachineProcessor.java [hunk 1]
```diff
@@ -251,6 +251,12 @@ public void updateProgressBar(@Nonnull BlockMenu inv, int slot, @Nonnull T opera
         // Update the progress bar in our inventory (if anyone is watching)
         int remainingTicks = operation.getRemainingTicks();
         int totalTicks = operation.getTotalTicks();
+
+        // If the operation is finished, we don't need to update the progress bar.
+        if (remainingTicks <= 0 && totalTicks <= 0) {
+            return;
+        }
+
         ChestMenuUtils.updateProgressbar(inv, slot, remainingTicks, totalTicks, getProgressBar());
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/machines/MachineProcessor.java:260]:
```suggestion
        // Fixes #3538 - If the operation is finished, we don't need to update the progress bar.
        if (remainingTicks > 0 || totalTicks > 0) {
            ChestMenuUtils.updateProgressbar(inv, slot, remainingTicks, totalTicks, getProgressBar());
        }
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
I approve of this, I even checked the logic and made a truth table.
But as pointed out, style-wise an inclusive-if would be preferred and we generally like to keep a reference to the fixed issue so we can better track these things.

Otherwise, it looks good! Thanks!

**Review by Sfiguz7 — APPROVED** [review]
LGTM

**Review by TheBusyBiscuit — APPROVED** [review]

**Review by svr333 — APPROVED** [review]

**Pull request merged**